status-im/status-desktop
# feat: optimize filter subscripitons

### What does the PR do

Underyling status-go changes as per https://github.com/status-im/status-go/pull/5440

### Affected areas

LightClient mode

### Test steps

To enable light-client, go to Settings --> Advanced and select "Light Mode". This will require an app restart.

1. Login normally, and make sure you enable Node Management and Debug in Advanced Settings.
2. Make sure you got some peers connected in Node Management tab
3. Run tests by being part of community , 1:1 chats, private group chats etc . Do wait for 30-60 seconds for all filter subscriptions to stabilize, until then you may notice a latency in receiving messages.
4. Check for number of filters installed by looking for below log (It gives an idea)
6. Disconnect from the internet, wait until the peer count goes to 0.
7. Connect to the internet, You should reconnect to some peers in sometime. (Note that it takes some time for peers to be connected - this needs fixing)
8. Test reception of messages on 1:1 chat, groups and community
9. Leave it running and ensure no issues in send/receive messages
10. tail geth.log and grep for "subs count". This indicates count of filter subscriptions. Note how long it takes to reach the max subscription count (it has been noticed that since peer connectivity fails to many discovered peers, it takes time)
